### ISSUE TITLE
Testability of HandleBasicDeliver in IAsyncBasicConsumer

### ISSUE THREAD

**Issue by Tornhoof** [rabbitmq/rabbitmq-dotnet-client#1630]
### Is your feature request related to a problem? Please describe.

Currently the method is defined as following:
HandleBasicDeliver in IAsynBasicConsumer is defined as following
https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/e247e8b14a90e50921c1a5f3bfa37f4c8036bfb1/projects/RabbitMQ.Client/client/api/IAsyncBasicConsumer.cs#L49-L55

Passing in ``ReadOnlyBasicProperties`` directly and not the interface ``ReadOnlyBasicProperties`` makes it hard to test in Unit Tests via mocks, the type is sealed and has no easy constructors.

I have a rather large suit of mocked tests of my derived types of DefaultAsyncBasicConsumer et. al. 

### Describe the solution you'd like

It would be nice if the method is either using the interface or ``ReadOnlyBasicProperties`` gets a constructor in the other direction from BasicProperties, just like the other way round.

### Describe alternatives you've considered

I'm currently using the wireformatting directly to create the type:
```csharp
        public static ReadOnlyBasicProperties FromBasicProperties(this BasicProperties basicProperties)
        {
            var writeable = (IAmqpWriteable)basicProperties;
            var bufferLength = writeable.GetRequiredBufferSize();
            var array = new byte[bufferLength];
            var length = writeable.WriteTo(array);
            var span = array.AsSpan(0, length);
            var readOnlyBasicProperties = new ReadOnlyBasicProperties(span);
            return readOnlyBasicProperties;
        }
```
This works but kinda feelds like a crutch.

**Comment by michaelklishin** [comment]
@Tornhoof if this is a matter of replacing a specific implementation type with an interface it implements, why not go straight to submitting a PR?

**Comment by Tornhoof** [comment]
> why not go straight to submitting a PR

done.

**Comment by lukebakken** [comment]
Fixed by #1631, thank you @Tornhoof
